src/MICore/LaunchOptions.cs

```diff
@@ -2079,6 +2106,25 @@ public interface IPlatformAppLauncherSerializer : IDisposable
         XmlSerializer GetXmlSerializer(string name);
     }
 
+    /// <summary>
+    /// Optionally used when implementing a launcher extention. The extention implements this interface in order to support json launch options
+    /// </summary>
+    [ComVisible(true)]
+    [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+    [Guid("B9EED3FB-4905-40B2-B865-6B4061E80A44")]
+    public interface IPlatformJsonAppLauncher : IDisposable
+    {
+        /// <summary>
+        /// Initializes the launcher from the json launch settings
+        /// </summary>
+        /// <param name="exePath">[Required] Path to the executable provided in the VsDebugTargetInfo by the project system. Some launchers may ignore this.</param>
+        /// <param name="args">[Optional] Arguments to the executable provided in the VsDebugTargetInfo by the project system. Some launchers may ignore this.</param>
+        /// <param name="dir">[Optional] Working directory of the executable provided in the VsDebugTargetInfo by the project system. Some launchers may ignore this.</param>
+        /// <param name="launcherJsonOptions">[Required] JObject options structure</param>
+        /// <param name="targetEngine">Indicates the type of debugging being done.</param>
+        void SetJsonLaunchOptions(string exePath, string args, string dir, object launcherJsonOptions, TargetEngine targetEngine);
```

Comment:
>object [](start = 75, length = 6)

This is always going to be JObject, correct? Should we strongly type it?

Reply:
Yes. Done.